libusbcommunication/src/main/java/me/jahnen/libaums/libusbcommunication/LibusbCommunication.kt

```diff
@@ -0,0 +1,114 @@
+package me.jahnen.libaums.libusbcommunication
+
+import android.content.ContentValues.TAG
+import android.hardware.usb.*
+import android.util.Log
+import com.github.mjdev.libaums.ErrNo
+import com.github.mjdev.libaums.usb.UsbCommunication
+import com.github.mjdev.libaums.usb.UsbCommunication.Companion.TRANSFER_TIMEOUT
+import com.github.mjdev.libaums.usb.UsbCommunicationCreator
+import java.io.IOException
+import java.nio.ByteBuffer
+import kotlin.math.absoluteValue
+
+class LibusbCommunication(
+        private val usbManager: UsbManager,
+        private val usbDevice: UsbDevice,
+        override val usbInterface: UsbInterface,
+        override val outEndpoint: UsbEndpoint,
+        override val inEndpoint: UsbEndpoint
+) : UsbCommunication {
+
+    // used to save heap address of libusb device handle
+    private var libUsbHandleArray = longArrayOf(0)
+    private val libUsbHandle: Long
+        get() = libUsbHandleArray[0]
+    private var deviceConnection: UsbDeviceConnection?
+
+    init {
+        System.loadLibrary("libusbcom")
+
+        deviceConnection = usbManager.openDevice(usbDevice)
+                ?: throw IOException("deviceConnection is null!")
+
+        if(!nativeInit(deviceConnection!!.fileDescriptor, libUsbHandleArray)) {
+            throw IOException("libusb init failed")
+        }
+
+//        val ret = nativeClaimInterface(libUsbHandle, usbInterface.id)
+//        if (ret < 0) {
+//            throw IOException("libusb returned $ret in claim interface")
+//        }
+        val claim = deviceConnection!!.claimInterface(usbInterface, true)
+        if (!claim) {
+            throw IOException("could not claim interface!")
+        }
```

Comment:
Why can't libusb claim it? I'm worried this may cause issues because I've seen it internally does unclaim/claim in reset_device, for example.